random-beans/src/test/java/io/github/benas/randombeans/PopulatorImplTest.java

```diff
@@ -145,6 +146,23 @@ public void javaNetTypesShouldBePopulated() throws Exception {
         assertThat(website.getUrl()).isNotNull();
     }
 
+    @Test(expected = BeanPopulationException.class)
+    public void failsToPopulateInterfacesAndAbstractClassesIfScanClasspathForConcreteClassesIsDisabled() throws Exception {
+        Populator nonScanningPopulator = aNewPopulatorBuilder().scanClasspathForConcreteClasses(false).build();
+
+        nonScanningPopulator.populateBean(Mamals.class);
+    }
+
+    @Test
+    public void generatesConcreteTypesForInterfacesAndAbstractClassesIfScanClasspathForConcreteClassesIsEnabled() throws Exception {
+        Populator scanningPopulator = aNewPopulatorBuilder().scanClasspathForConcreteClasses(true).build();
+
+        Mamals mamals = scanningPopulator.populateBean(Mamals.class);
+
+        assertThat(mamals.getMamal()).isOfAnyClassIn(Human.class, Ape.class, Person.class, SocialPerson.class);
+        assertThat(mamals.getMamalImpl()).isOfAnyClassIn(Human.class, Ape.class, Person.class, SocialPerson.class);
+    }
+
```

Comment:
I'm curious about how the new code will behave if I have a field of type `Comparable` in my bean:

```java
class Person {
   private Comparable c1;
   private Comparable<Date> c2;
}
```

There are hundreds of comparable implementations out there (635 in the classpath of the project) and I would like to know how the code will behave if the implementation that has been chosen randomly is abstract, like `sun.org.mozilla.javascript.internal.ast.AstNode` or `com.sun.tools.javadoc.ProgramElementDocImpl`.. Do you see the trick? Can you please add a test case with a `Comparable` field? Many thanks